app/logic/userManagement.py

```diff
@@ -33,3 +33,27 @@ def changeCurrentTerm(term):
     newCurrentTerm.save()
 
     session["current_term"] = model_to_dict(newCurrentTerm)
+
+def addTerm():
+    terms_table = {"Spring":"Summer",
+                    "Summer":"Fall",
+                    "Fall":"Spring"}
+    terms = list(Term.select().order_by(Term.id))
+    lastCreatedTerm = terms[-1]
+    termDescription = lastCreatedTerm.description
+    term = termDescription.split()
+    year = termDescription[-1]
+    if termDescription[0]=="Fall": # We only change the year when it is Fall
+        year = int(year) + 1
```

Comment:
This logic won't work if we're going from, say, 2029 to 2030, will it? You're only getting a single digit off of termDescription, not the whole year. The whole year is `term[1]`, not `termDescription[-1]`, if I'm reading it correctly

Reply:
The problem was with the variable term. It was not used correctly. We have fixed and the logic stays the same and should work.